### ISSUE TITLE
vl2png yields ESPIPE read error.

### ISSUE THREAD

**Issue by VTGuy08** [vega/vega-lite#5510]
Environment: 
Mac 10.14
NodeJS 9.8.0

When using the 'vl2png' script, I receive an error: 
{ Error: ESPIPE: invalid seek, read errno: -29, code: 'ESPIPE', syscall: 'read' }

Command used: ./vl2png test_spec.json out.png

I was able to debug the issue and it seems that '/dev/stdin' is being passed into the read.js file of the vega-cli instead of the actual json spec.  I know this because if I go into read.js (line 7) and modify it such that const input = process.stdin;  Then I am able to generate a PNG file.

I kindly request that you correct/update the vl2png scripts (unless I'm doing something wrong xD ).

test_spec.json:
```json
{"schema":"https://vega.github.io/schema/vega-lite/v4.json","padding":5,"transform":[],"data":{"url":"https://vega.github.io/editor/data/population.json"},"background":"green","width":300,"title":{"offset":4,"anchor":"middle","fontSize":16,"text":"New Title","frame":"group"},"layer":[{"encoding":{"color":{"field":"age","scale":{"scheme":"viridis"}},"x":{"field":"people","type":"quantitative","axis":{"grid":true,"labels":true},"aggregate":"sum"},"y":{"field":"age","type":"ordinal","axis":{"grid":false,"labels":true}}},"mark":{"tooltip":true,"type":"bar"}},{"encoding":{"x":{"field":"people","type":"quantitative","axis":{"grid":true,"labels":true},"aggregate":"sum"},"y":{"field":"age","type":"ordinal","axis":{"grid":false,"labels":true}},"text":{"field":"people","type":"quantitative","axis":{"grid":true,"labels":true},"aggregate":"sum"}},"mark":{"dx":30,"dy":0,"tooltip":false,"angle":0,"dir":"ltr","type":"text"}}],"height":300,"autosize":"pad"}

```

Please:

- [X ] Check for duplicate issues. Please file separate requests as separate issues on GitHub.
- [X ] Describe how to reproduce the bug / the goal of the feature request.
- [X ] Use the latest versions of Vega and Vega-Lite.
- [ X] Provide an example spec in JSON, wrapped by triple backticks like this:

```json
{
  "mark": "point",
  "encoding": {"x": {"field": "a"}}
}
```

- [X ] If applicable, include error messages and screenshots, GIF videos (e.g. using https://www.cockos.com/licecap/), or working examples (e.g. by clicking share in the [Vega-Editor](https://vega.github.io/editor/) or https://bit.ly/vega-lite-blocks)

**Comment by domoritz** [comment]
Thank you for filing the issue. Could you send us a pull request to fix the issue?

**Comment by wessven** [comment]
I'm getting the same issue.  On Bash, if I run vl2svg I get (Example 1)

    $ vl2svg vega-test-graph2.json
    { Error: ESPIPE: invalid seek, read errno: -29, code: 'ESPIPE', syscall: 'read' }

The following works as expected (Example 2)

    $ vl2vg vega-test-graph2.json > test.vg
    $ vg2svg test.vg > output.svg

But this does not (Example 3)

    $ vg2svg <(vl2vg vega-test-graph2.json)
    { Error: ESPIPE: invalid seek, read errno: -29, code: 'ESPIPE', syscall: 'read' }

The issue occurs when render.js calls into read.js.  I can see that render.js passes in a string in both the first and third examples.  For the first it passes in `/dev/stdin` and in the third `/dev/fd/63`.  What this tells me, is that in read.js, the following condition will never assign `process.stdin`:

```javascript
const input = file ? fs.createReadStream(file) : process.stdin;
```

As with @VTGuy08, if I assign `process.stdin` directly to `input`, I can make it work for the first example.  But then, of course, the second example breaks.  The third example still doesn't work.

I'm quite green with Node.js, so don't have a solution to propose at this time.

**Comment by wessven** [comment]
It seems like the following works for all three examples:

```javascript
const fs = require('fs');
const stream = require('stream')

module.exports = (file, encoding='utf8') => {
  return new Promise((resolve, reject) => {
    const input = new stream.PassThrough();
    input.write(fs.readFileSync(file).toString());
    input.end();
    let text = '';

    input.setEncoding(encoding);
    input.on('error', err => { reject(err); });
    input.on('data', chunk => { text += chunk; });
    input.on('end', () => { resolve(text); });
  });
};
```

Probably not the most efficient code, but it works.  Don't think more is required for this application.  Should I submit a pull request?

**Comment by domoritz** [comment]
I am unable to reproduce the error. Can you give me the exact steps you used to install vega and vega-lite and the spec you use?

`vl2vg examples/specs/bar.vl.json | vg2svg` works for me after `npm install -g vega-lite vega`.

**Comment by wessven** [comment]
I'm using Node 8.10, because that is what is available on Ubuntu 18.04.  Could it be that?

**Comment by domoritz** [comment]
Node 8 is deprecated so we won’t support it.
